test/ui_test_suite.rb

```diff
@@ -3504,6 +3515,234 @@ def open_new_page(url)
 		assert (plot_labels.include? "user-#{$random_seed}: group0 (3 points)"), "labels are incorrect: '#{plot_labels}' should include 'user-#{$random_seed}: group0'"
 	end
 
+	# make sure sharing the annotation works
+	test 'front-end: user annotation sharing' do
+		puts "Test method: #{self.method_name}"
+
+		# login
+		login_path = @base_url + '/users/sign_in'
+		@driver.get login_path
+		wait_until_page_loads(login_path)
+		login($test_email)
+
+		# load annotation panel
+		annot_path = @base_url + '/user_annotations'
+		@driver.get annot_path
+
+		@driver.find_element(:class, "user-#{$random_seed}-exp-edit").click
+		wait_until_page_loads('edit user annotation path')
+
+		# click the share button
+		share_button = @driver.find_element(:id, 'add-user-annotation-share')
+		share_button.click
+
+		share_email = @driver.find_element(:class, 'share-email')
+		share_email.send_keys($share_email)
+
+		share_permission = @driver.find_element(:class, 'share-permission')
+		share_permission.send_keys('Edit')
+
+		#update the annotation
+		submit = @driver.find_element(:id, 'submit-button')
+		submit.click
+
+		wait_until_page_loads('user annotation path')
+
+		# logout
```

Comment:
It's actually not necessary to call wait_for_render before closing modals anymore as close_modal does this now.